### PR TITLE
Use default network for dpdk test

### PR DESCRIPTION
**Pull request by SchSeba**
This PR change to dpdk test to create a policy when the test are not in discovery mode to find and use the default router and use that interface.

I make this change because on the default network there is always
traffic for the testpmd application to process, this makes the test stable.

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

File: functests/dpdk/dpdk.go [hunk 1]
```diff
@@ -581,6 +581,68 @@ func ValidateSriovPolicy() (bool, error) {
...
+		if pod.Spec.NodeName == nodeName {
+			return pod, nil
+		}
+	}
+
+	return corev1.Pod{}, fmt.Errorf("failed to find pod on node %s", nodeName)
+}
+
+// FindDpdkSriovDevice will search for the default gateway interface to be used as the PF
+func FindDpdkSriovDevice(enabledNodes *sriovcluster.EnabledNodes, nodeName string) (*sriovv1.InterfaceExt, error) {
```
Comment (fedepaol) [inline functests/dpdk/dpdk.go:595]: Does this need to be exported?

File: functests/dpdk/dpdk.go [hunk 2]
```diff
@@ -581,6 +581,68 @@ func ValidateSriovPolicy() (bool, error) {
...
+	s, ok := enabledNodes.States[nodeName]
```
Comment (fedepaol) [inline functests/dpdk/dpdk.go:596]: nodeStatus. You are using s way below, where it's harder to get its context

File: functests/dpdk/dpdk.go [hunk 3]
```diff
@@ -581,6 +581,68 @@ func ValidateSriovPolicy() (bool, error) {
...
+	if !ok {
+		return nil, fmt.Errorf("Node %s not found", nodeName)
+	}
+
+	podList := &corev1.PodList{}
+	err := client.Client.List(context.Background(), podList, &goclient.ListOptions{Namespace: SRIOV_OPERATOR_NAMESPACE,
+		LabelSelector: labels.SelectorFromSet(labels.Set{"app": "sriov-network-config-daemon"})})
```
Comment (fedepaol) [inline functests/dpdk/dpdk.go:603]: Can't you just use FieldsSelector here and avoid the findPodBy..?

File: functests/dpdk/dpdk.go [hunk 4]
```diff
@@ -414,7 +414,7 @@ func findOrOverrideSriovPolicyAndNetwork() {
 
 		Expect(len(nn)).To(BeNumerically(">", 0))
 
-		sriovDevice, err := sriovInfos.FindOneSriovDevice(nn[0])
+		sriovDevice, err := FindDpdkSriovDevice(sriovInfos, nn[0])
```
Comment (fedepaol) [inline functests/dpdk/dpdk.go:417]: Do we need the same change in discovery mode? See https://github.com/openshift-kni/cnf-features-deploy/blob/4c169cd5bdce2a098c0923934917395ca4fe2985/functests/utils/discovery/discovery.go#L69

File: functests/dpdk/dpdk.go (hunk 1 above)
Comment (SchSeba) [reply functests/dpdk/dpdk.go]: no need

File: functests/dpdk/dpdk.go (hunk 2 above)
Comment (SchSeba) [reply functests/dpdk/dpdk.go]: +1

File: functests/dpdk/dpdk.go (hunk 3 above)
Comment (SchSeba) [reply functests/dpdk/dpdk.go]: +1

File: functests/dpdk/dpdk.go (hunk 4 above)
Comment (SchSeba) [reply functests/dpdk/dpdk.go]: No I want to leave the user the ability to test on a different interfaces that are not the default gateway interface.

In general in discovery mode the user should know why he creates the policy on a specific interface.

Comment (fedepaol) [reply functests/dpdk/dpdk.go]: Yep, but even in discovery mode there will be interfaces belonging to the pf where the default gw is, and others belonging to other pfs. Would it make sense to favour the first one?

Comment (SchSeba) [reply functests/dpdk/dpdk.go]: Let me fix the discovery in a separate PR, I need to cherry-pick this to release 4.5

Comment (fedepaol) [reply functests/dpdk/dpdk.go]: Agree

**Pull request merged**